### PR TITLE
Users

### PR DESCRIPTION
**Pull request by mateenqazi**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
```
Comment (mateenqazi) [inline .vscode/launch.json:3]: add this file into gitignore

File: main.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	db := config.DatabaseConnection()
+	validate := validator.New()
+
+	ginRouter := gin.Default()
+
+	// http.RegisterRoutes(ginRouter, db, validate)
+	userRoutes.RegisterRoutes(ginRouter, db, validate)
+
+	server := &http.Server{
+		Addr:    ":8000",
```
Comment (mateenqazi) [inline main.go]: when you add .env file. then, port coming from .env

File: users/domain/user/repository.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ type Repository interface {
 }
 
 type ReadRepository interface {
```
Comment (mateeullahmalik) [inline users/domain/user/repository.go:14]: pass the context here
domain.Users

File: users/domain/user/repository.go [hunk 4]
```diff
@@ -12,6 +12,7 @@ type Repository interface {
...
+	GetAll() ([]domain.User, error)
```
Comment (mateenqazi) [inline users/domain/user/repository.go:15]: user domain.Users
pass context here.

File: users/infrastructure/postgres/users.go [hunk 5]
```diff
@@ -86,12 +86,32 @@ func (u UserPostgres) Save(ctx context.Context, request domain.User) (domain.Use
 
 	user := User{}.FromDomain(request)
 
+	// var existingUser User
```
Comment (mateenqazi) [inline users/infrastructure/postgres/users.go:89]: use seperate function for validation

**Pull request merged**